roles/openshift_health_checker/openshift_checks/efk_stack_check.py

```diff
@@ -0,0 +1,274 @@
+            results['free'] = int(0)
+
+        return results
+
+    def check_fluentd(self, task_vars):
+        """ Verify fluentd is running """
+        fluentd_status = {}
+
+        # Get all nodes with fluentd label
+        nodes = json.loads(self.oc("get nodes -o json", task_vars))
+        fluentd_nodes = []
+        for node in nodes['items']:
+            if 'logging-infra-fluentd' in node['metadata']['labels']:
+                if node['metadata']['labels']['logging-infra-fluentd'] == 'true':
+                    fluentd_nodes.append(node)
+
+        # Make sure fluentd is on all the nodes and the pods are running
+        fluentd_status['number_expected_pods'] = len(fluentd_nodes)
+        fluentd_status['number_pods'] = len(self.fluentd_pods)
+        fluentd_status['node_mismatch'] = 0
+        fluentd_status['running'] = 1
+
+        for pod in self.fluentd_pods:
+            node_matched = False
+
+            if pod['status']['containerStatuses'][0]['ready'] is False:
+                fluentd_status['running'] = 0
+
+            # If there is already a problem don't worry about looping over the remaining pods/nodes
+            for node in fluentd_nodes:
+                internal_ip = ""
+                for address in node['status']['addresses']:
+                    if address['type'] == "InternalIP":
+                        internal_ip = address['address']
+
+                try:
+                    if node['metadata']['labels']['kubernetes.io/hostname'] == pod['spec']['host']:
+                        node_matched = True
+                        break
+
+                    raise ValueError('')
+                except:
+                    if internal_ip == pod['spec']['nodeName'] or node['metadata']['name'] == pod['spec']['nodeName']:
+                        node_matched = True
+                        break
+
+            if node_matched is False:
+                fluentd_status['node_mismatch'] = 1
+                break
+
+
+        return fluentd_status
+
+    def check_kibana(self, task_vars):
+        """ Check to see if kibana is up and working """
+        kibana_status = {}
+
+        # Get logging url
+        get_route = self.oc("get route logging-kibana -o json", task_vars)
+        if not get_route:
+            return kibana_status
+
+        route = json.loads(get_route).get("status", {}).get("ingress", [])[0]
+        if not route:
+            return kibana_status
+
+        route = route.get("host", "")
+        kibana_url = "https://{}/".format(route)
+
+        # Disable SSL to work around self signed clusters
+        ctx = ssl.create_default_context()
+        ctx.check_hostname = False
+        ctx.verify_mode = ssl.CERT_NONE
+
+        # Verify that the url is returning a valid response
+        kibana_status['site_up'] = 1
+        try:
+            # We only care if the url opens
+            urllib2.urlopen(kibana_url, context=ctx)
+
+        except urllib2.HTTPError, e:
```

Comment:
Use `as` instead of `,`.

That's the Python 2 / Python 3 way. Comma is legacy syntax.